lib/govspeak/html_sanitizer.rb

```diff
@@ -60,17 +60,22 @@ def sanitize(allowed_elements: [])
   end
 
   def sanitize_config(allowed_elements: [])
+    # We purposefully disable style elements which Sanitize::Config::RELAXED allows
+    elements = Sanitize::Config::RELAXED[:elements] - %w[style] +
+      %w[govspeak-embed-attachment govspeak-embed-attachment-link svg path].concat(allowed_elements)
+
     Sanitize::Config.merge(
       Sanitize::Config::RELAXED,
-      elements: Sanitize::Config::RELAXED[:elements] + %w[govspeak-embed-attachment govspeak-embed-attachment-link svg path].concat(allowed_elements),
+      elements: elements,
       attributes: {
-        :all => Sanitize::Config::RELAXED[:attributes][:all] + %w[role aria-label],
+        # We purposefully disable style attributes which Sanitize::Config::RELAXED allows
+        :all => Sanitize::Config::RELAXED[:attributes][:all] + %w[role aria-label] - %w[style],
         "a" => Sanitize::Config::RELAXED[:attributes]["a"] + [:data] + %w[draggable],
-        "svg" => Sanitize::Config::RELAXED[:attributes][:all] + %w[xmlns width height viewbox focusable],
-        "path" => Sanitize::Config::RELAXED[:attributes][:all] + %w[fill d],
+        "svg" => %w[xmlns width height viewbox focusable],
+        "path" => %w[fill d],
         "div" => [:data],
-        # @TODO  These style attributes can be removed once we've checked there
-        # isn't hardcoded HTML in documents that uses them
+        # The style attributes are permitted here just for the ones Kramdown for table alignment
+        # we replace them in a post processor.
         "th" => Sanitize::Config::RELAXED[:attributes]["th"] + %w[style],
         "td" => Sanitize::Config::RELAXED[:attributes]["td"] + %w[style],
```

Comment:
Am I right in thinking this means a user could still provide arbitrary styles on `<th>` and `<td>` elements? And are we okay with that?

For example, would this work?

```markdown
## Some heading

<table>
  <tr>
    <td style="font-family: 'Comic Sans MS'; color: pink;">
      Will this work?
    </td>
  </tr>
</table>
```

Handily, it looks like the Sanitize gem allows us sanitize CSS properties too. The defaults provided by [Sanitize::Config::RELAXED](https://github.com/rgrove/sanitize/blob/a92f21cd223a32a1737262d68e56a4fb8b9470f9/lib/sanitize/config/relaxed.rb#L39-L721) are quite permissive – so perhaps we'd want to lock this down to just `text-align`?

e.g.

```ruby
css: {
  properties: %w[text-align]
}

Reply:
No, it's not that arbitrary styles are allowed on those elements. There is some pre-existing code (see below) that removes the style attribute of td and th elements if they have a style value that isn't text-align: `(left|center|right)`. This is what convinced me that the style ability was a bug as someone had carefully set-up regex's to restrict access.
https://github.com/alphagov/govspeak/blob/907314932068b68f18e6d8020fb03d35700241f4/lib/govspeak/html_sanitizer.rb#L20-L36, 

It's a good shout about the sanitize config though, I wonder if that would be cleaner than the ruby we have - I'll have a quick experiment.

Reply:
I've had a look and can apply it. It does clean things up, but I ended up with a bit of churn along the way. I think it's still a net improvement. Let me know what you think.

Reply:
@kevindew nice! 🎉  I definitely think this simplifies the implementation. Thanks!